### PR TITLE
Add recipe for libtree

### PR DESCRIPTION
**Pull request by awvwgk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
Macos probably needs a higher SDK version (which can only be done after creating the actual feedstock?).
Win was missing the elfio package, probably the file had not propagated to the mirror yet.

Apart from that, the package looks good!

**Comment by matthiasdiener** [comment]
I'd just set `skip=True` for MacOS and Win on this initial version of the package and deal with the other archs later.

Also, please feel free to add me as a maintainer if you like.

File: recipes/libtree/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+about:
+  home: https://github.com/haampie/libtree
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: ldd as a tree with an option to bundle dependencies into a single folder
+
+extra:
+  recipe-maintainers:
+    - awvwgk
```
Comment (awvwgk) [inline recipes/libtree/meta.yaml:45]:
```suggestion
    - awvwgk
    - matthiasdiener
```

File: recipes/libtree/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/haampie/libtree/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 099e85d8ba3c3d849ce05b8ba2791dd25cd042a813be947fb321b0676ef71883
+
+build:
+  number: 0
```
Comment (awvwgk) [inline recipes/libtree/meta.yaml:13]:
```suggestion
  number: 0
  skip: true  # [not linux]
```

**Comment by awvwgk** [comment]
@conda-forge/staged-recipes This recipe is ready for review. Linux only for now.

**Pull request merged**